packages/react-dom/src/test-utils/FizzTestUtils.js

```diff
@@ -133,6 +139,86 @@ async function replaceScriptsAndMove(
   }
 }
 
+export function installScriptObserver(window: any, document: Document) {
```

Comment:
@mofeiZ I was having trouble getting the latest technique of delaying bootstrap scripts to work with the current test setup for moving scripts. The problem is that in an async tick a script is added to the body but there is no visibility into this event for us to use the trick we use in `act`.

I implemented an entirely different approach where we use a mutation observer and look for unexecuted scripts on each new node addition. I believe this approach can work for all prior and new use cases but I may not understand well why mutation observer wasn't used in the first place so I would love a review here especially to see if I may need to make changes.

I took this opportunity to also make `act` better. It now infers the insertion point based on what is being streamed in so we don't need `actIntoEntireDocument`. I renamed the original `act` to `actIntoContainer`. it's not representative of 99% of how streaming will be done so I made it not the default name and we should probably move away from using it